astropy/coordinates/baseframe.py

```diff
@@ -682,7 +729,11 @@ def __bool__(self):  # Py 3.x
 
     @property
     def shape(self):
-        return self.data.shape
+        return self.data.shape if self.has_data else self._no_data_shape
```

Comment:
The reason I implemented it like this is that in the frame attribute I look up the frame shape, and I felt the attribute should not have to worry about whether that frame has data or not. 

More generally, it also seemed there should be a way to tell what shape the output would have if you transformed, say, a single coordinate to a frame with multiple times (say, an AltAz frame). With that, it is also clearer what to expect when you reshape or ravel a data-less frame.

But I do agree it is slightly counterintuitive to get an `AttributeError` when you access `size` and not when you access `shape`. Clearly, this is something that I should note explicitly in the to-be-written documentation!